presentation-compiler/src/main/scala/org/alephium/ralph/lsp/pc/sourcecode/SourceCodeState.scala

```diff
@@ -67,11 +67,14 @@ object SourceCodeState {
 
   }
 
+  /** Represents: Code that is parsed and compiled. */
+  sealed trait IsParsedAndCompiled extends IsCodeAware
+
   /** Represents: Code that is parsed. */
-  sealed trait IsParsed extends IsCodeAware
+  sealed trait IsParsed extends IsParsedAndCompiled
```

Comment:
It's a bit confusing here

A parsed code isn't necessary compiled, but with the types here, we read that if a code `IsParsed` it's also compiled as it extends `IsParsedAndCompiled`

Reply:
> A parsed code isn't necessary compiled ...

You are right. 

Would naming it `IsParsedOrCompiled` make it clearer? `IsParsedAndCompiled` is a bigger set (co-domain), and `IsParsed` is a subset (range) within that bigger set. The set `IsParsedAndCompiled` can contain `IsParsed`, `IsCompiled` or both.

For example:

```scala
sealed trait Mac extends OS
sealed trait Linux extends OS
// or
sealed trait Mac extends MacOrLinux
sealed trait Linux extends MacOrLinux
```

`IsParsedAndCompiled` was used because this name matches the function [`parseAndCompile(workspace)`](https://github.com/alephium/ralph-lsp/blob/master/presentation-compiler/src/main/scala/org/alephium/ralph/lsp/pc/workspace/Workspace.scala#L318).

Reply:
Ok got it, `IsParsedOrCompiled` is indeed a bit clearer when reading the code, while not having the big picture, but it makes sense when you explain. 
Maybe you could just add a very small comment above `IsParsedAndCompiled` just saying it's the co-domain for parsed and compiled?